### PR TITLE
CIRCLE-3360 building custom images

### LINKED ISSUE #726: Docker Executor Base Image Requirements
I've noticed the following requirements so far:

Command/Tool
- git (checkout step fails without it)
- ssh
- CA certificates (cache save/restore fails without it)

Corresponding Packages:
- alpine
  - git
  - openssh
  - ca-certificates

We should probably document this somewhere.

Optional
- many people use Tar/GZIP when using our cache. Alpine doesn't have it
-Alpine packages
  - tar
  - gzip

### PR DESCRIPTION
**Pull request by smaant**
Fixes #726

### CODE REVIEW

File: jekyll/_cci2/custom-images.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+layout: classic-docs
+title: "Building Custom Image for Docker Executor"
```
Comment (felicianotech) [inline jekyll/_cci2/custom-images.md]: Should be "Building a Custom Image for Docker Executor" or "Building Custom Images for Docker Executor".

**Comment by felicianotech** [comment]
This is nice. I linked this PR to my Issue I opened last week.

File: jekyll/_cci2/custom-images.md (hunk 1 above)
Comment (smaant) [inline jekyll/_cci2/custom-images.md]: I think "Images" is better, will change, thank you!

**Comment by keybits** [comment]
This is great, thank you @smaant :-)

I made some grammar and clarity edits, otherwise, LGTM.

File: jekyll/_cci2/custom-images.md [hunk 2]
```diff
@@ -6,48 +6,49 @@ categories: [configuring-jobs]
...
 ---
 
-Since CircleCI 2.0 platform is built with a flexibility of Docker in mind the best way to use this power is to create custom images for your jobs. There are few advantages for doing this:
- 1. Image will include all required tools that would eliminate need to install them at each run therefore reduce job execution time
- 2. Config file will be more concise and will contain only relevant to the job purpose steps
+CircleCI 2.0 gives you access to the power and flexibility of Docker. One of the ways you can take advantage of this is to create custom Docker images for your jobs. The benefits of doing this are:
 
-In this document we will give a walkthrough how to create a custom image. In most cases you'll want to have a custom image for your [primary container]({{ site.baseurl }}/2.0/glossary/#primary-container) so we'll mostly describe this case. But you can easily apply this knowledge for creating images for supporting containers as well.
+1. Faster job execution because you can include all tools you require, eliminating the need to install them on each job run
+2. A more concise and and easier to maintain CircleCI `config.yml` file
```
Comment (smaant) [inline jekyll/_cci2/custom-images.md]: "and and"

File: jekyll/_cci2/custom-images.md [hunk 3]
```diff
@@ -6,48 +6,49 @@ categories: [configuring-jobs]
 order: 30
 ---
 
-Since CircleCI 2.0 platform is built with a flexibility of Docker in mind the best way to use this power is to create custom images for your jobs. There are few advantages for doing this:
- 1. Image will include all required tools that would eliminate need to install them at each run therefore reduce job execution time
- 2. Config file will be more concise and will contain only relevant to the job purpose steps
+CircleCI 2.0 gives you access to the power and flexibility of Docker. One of the ways you can take advantage of this is to create custom Docker images for your jobs. The benefits of doing this are:
 
-In this document we will give a walkthrough how to create a custom image. In most cases you'll want to have a custom image for your [primary container]({{ site.baseurl }}/2.0/glossary/#primary-container) so we'll mostly describe this case. But you can easily apply this knowledge for creating images for supporting containers as well.
+1. Faster job execution because you can include all tools you require, eliminating the need to install them on each job run
```
Comment (smaant) [inline jekyll/_cci2/custom-images.md]: I think "preinstall" might have more meaning here than "include"

File: jekyll/_cci2/custom-images.md [hunk 4]
```diff
@@ -6,48 +6,49 @@ categories: [configuring-jobs]
...
 
 ## Base image
 
-First of all you need to choose a base image. In general it makes sense to use image with your main language/framework as a base images. [Docker Hub](https://hub.docker.com/) has tons of images for almost all popular languages and frameworks. We recommend to start from [officially supported images](https://hub.docker.com/explore/).
+First of all you need to choose a base image. In general it makes sense to use a image with your main language/framework as a base images. [Docker Hub](https://hub.docker.com/) has pre-built images for most popular languages and frameworks. We recommend starting with an [officially supported image](https://hub.docker.com/explore/).
 
-Once base image is chosen we can start writing `Dockerfile`:
+Once you've chosen abase image you can start writing a `Dockerfile` to extend it:
```
Comment (smaant) [inline jekyll/_cci2/custom-images.md]: "a base" probably

**Comment by keybits** [comment]
@smaant thanks for spotting those typos. Updated and feel free to merge if you're happy.

**Pull request merged**